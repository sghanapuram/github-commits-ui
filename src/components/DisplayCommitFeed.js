import React from 'react';

const DisplayCommitFeed = ({ data }) => (
    <table className="commits-table">
        <thead>
            <tr>
                <th>Timestamp</th>
                <th>Commit</th>
                <th>Author</th>
            </tr>
        </thead>
        <tbody>
            {
                data.map((item, i) => (
                    <tr key={ i }>
                        <td className="date">
                            { item.commit.author.date }
                        </td>
                        <td className="message">
                            <a
                                className="commit-link"
                                href={ item.html_url }
                                target="_blank"
                                rel="noreferrer"
                            >
                                { item.commit.message }
                            </a>
                        </td>
                        <td className="author">
                            { item.commit.author.name }
                        </td>
                    </tr>
                ))
            }
        </tbody>
    </table>
);

export default DisplayCommitFeed;
