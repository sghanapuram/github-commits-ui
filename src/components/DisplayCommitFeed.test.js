import { render, screen } from '@testing-library/react';
import DisplayCommitFeed from './DisplayCommitFeed';

const response =  [
    {
      commit: {
        author: {
          date: '05-07-2021',
          name: 'name'
        },
        message: 'test'
      },
      html_url: 'test_url'
    }
];

let container;

test('renders Commit feed table', () => {
    container = render(<DisplayCommitFeed data={ response } />);
    expect(container).toMatchSnapshot();
});

test('renders commits table header', () => {
    container = render(<DisplayCommitFeed data={ response } />);
    const timestamp = screen.getByText('Timestamp');
    const message = screen.getByText('Commit');
    const author = screen.getByText('Author');
    expect(timestamp).toBeInTheDocument();
    expect(message).toBeInTheDocument();
    expect(author).toBeInTheDocument();
});


test('renders commits table data', () => {
    container = render(<DisplayCommitFeed data={ response } />);
    const timestamp = screen.getByText('05-07-2021');
    const message = screen.getByText('test');
    const author = screen.getByText('name');
    expect(timestamp).toBeInTheDocument();
    expect(message).toBeInTheDocument();
    expect(author).toBeInTheDocument();
});

