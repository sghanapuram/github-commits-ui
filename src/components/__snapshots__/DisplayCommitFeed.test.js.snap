// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Commit feed table 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="commits-table"
      >
        <thead>
          <tr>
            <th>
              Timestamp
            </th>
            <th>
              Commit
            </th>
            <th>
              Author
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="date"
            >
              05-07-2021
            </td>
            <td
              class="message"
            >
              <a
                class="commit-link"
                href="test_url"
                rel="noreferrer"
                target="_blank"
              >
                test
              </a>
            </td>
            <td
              class="author"
            >
              name
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="commits-table"
    >
      <thead>
        <tr>
          <th>
            Timestamp
          </th>
          <th>
            Commit
          </th>
          <th>
            Author
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="date"
          >
            05-07-2021
          </td>
          <td
            class="message"
          >
            <a
              class="commit-link"
              href="test_url"
              rel="noreferrer"
              target="_blank"
            >
              test
            </a>
          </td>
          <td
            class="author"
          >
            name
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
